src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -218,6 +222,7 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context,
             // create a bitset of ordinals corresponding to the rows in the given key range
             SparseFixedBitSet bits = bitSetForSearch();
             boolean hasMatches = false;
+            InMemoryRowIdsView inMemoryRowIdsView = null;
```

Comment:
Is there a comment somewhere that explains the design here and how this helps solve the shadowed key problem?

Reply:
There isn't. Since we're likely not going in this direction, it might not get added. Here is some brief motivation.

When we hit this code block, we have a range query and we do a reverse lookup from segment id to ordinal id to get the ordinals in the range. This `InMemoryRowIdsView` stores the mapping of `ordinal to row id` such that all row ids are known to be not shadowed. As a side effect, we also get to skip some disk reads.